java/lib.jshell.agent/build.xml

```diff
@@ -55,20 +55,6 @@
     </target>
 
     <target name="jar-probe" depends="probe-init">
-        <unjar src="${agentsrc.jshell.cp}" dest="build/agent/classes">
-            <patternset>
-                <include name="jdk/jshell/execution/**"/>
-                <exclude name="**/JDI*.class"/>
-                <exclude name="**/LocalExecutionControl.class"/>
-                <exclude name="**/StreamingExecutionControl.class"/>
-                <!--<exclude name="**/DemultiplexInput.class"/>-->
-            </patternset>
-            <patternset>
-                <include name="jdk/jshell/spi/ExecutionControl*.class"/>
-                <exclude name="jdk/jshell/spi/ExecutionControl*Generator.class"/>
-                <include name="jdk/jshell/spi/SPIResolutionException.class"/>
-            </patternset>
-        </unjar>
```

Comment:
couldn't figure out why things like this existed so I looked through git history. NB bundled jshell before it was part of JDK 9 https://github.com/emilianbold/netbeans-releases/blob/7c1a0f7ed55b753310aeff2d1fbb45e996ebb7b1/libs.jshell/src/org/netbeans/libs/jshell/Bundle.properties#L44

But what remained of it is that https://netbeans.osuosl.org/binaries/1CA9DE21C6E443E814CB56912BEE872EF7094C49-jshell-9.jar is used to build the agent against, so removing it is fine since it is all in the JDK now, right?

Reply:
This code bundles parts of the JDK into the agent `jar`. The build of `java/lib.jshell.agent` primarily produces two artifacts:

- `nbbuild/netbeans/java/modules/ext/nb-custom-jshell-probe.jar`
- `nbbuild/netbeans/java/modules/ext/nb-mod-jshell-probe.jar`

When you compare the two files before your change, you'll notice the two packages `jdk.jshell.execution` and `jdk.jshell.spi`. These two are only present in the `nb-custom-jshell-probe.jar` file. After your change the two files are identical and don't anymore for JDK8. I tested with master, there running on JDK17 I can call a JShell for 8, with your change that fails. I did not check further, but I assume, that JDK8 does not carry the JShell code  and with `jshell-9.jar` removed, the agent can't be started.

Reply:
makes sense. I might revert this to the original version https://github.com/apache/netbeans/commit/5a05442bdd1d9caf1ae878f007fcc78638f67660 and play with it a bit more.

Reply:
current version has now `nb-mod-jshell-probe.jar` and references to it removed. Still not sure what approach to take but there is no rush.